src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -367,7 +378,61 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
         container.pendingContext = context;
       }
 
-      scheduleTopLevelUpdate(current, element, callback);
+      scheduleTopLevelUpdate(current, element, false, callback);
+    },
+
+    // Like updateContainer, but blocks the root from committing. Returns an
+    // expiration time.
+    // TODO: Can this be unified with updateContainer? Or is it incompatible
+    // with the existing semantics of ReactDOM.render?
+    updateRoot(
+      element: ReactNodeList,
+      container: OpaqueRoot,
+      parentComponent: ?React$Component<any, any>,
```

Comment:
If this is an new API we don't have to support legacy apis like renderSubtreeIntoContainer.